pkg/controller/logcollector/logcollector_controller.go

```diff
@@ -389,6 +389,13 @@ func (r *ReconcileLogCollector) Reconcile(ctx context.Context, request reconcile
 		return reconcile.Result{}, err
 	}
 
+	// eksLogForwarderKeyPair is the key pair eks-log-forwarder presents to identify itself
+	eksLogForwarderKeyPair, err := certificateManager.GetOrCreateKeyPair(r.client, render.EKSLogForwarderTLSSecretName, common.OperatorNamespace(), []string{render.EKSLogForwarderTLSSecretName})
```

Comment:
I think we should only do this inside of the `	if instance.Spec.AdditionalSources.EksCloudwatchLog != nil {` branch below on line 545. 

No need to create this secret if we're not running on EKS, and EKS log forwarding isn't enabled.